test/e2e/network/service.go

```diff
@@ -883,13 +886,23 @@ var _ = common.SIGDescribe("Services", func() {
 		podname1 := "pod1"
 		podname2 := "pod2"
 
-		createPodOrFail(f, ns, podname1, jig.Labels, containerPorts1)
+		args1 := []string{"nettest", "--port", fmt.Sprintf("%d", port1)}
+		container1 := e2epod.NewAgnhostContainer("agnhost-container-1", nil, containerPorts1, args1...)
+
+		args2 := []string{"nettest", "--port", fmt.Sprintf("%d", port2)}
+		container2 := e2epod.NewAgnhostContainer("agnhost-container-2", nil, containerPorts2, args2...)
+
+		pod := createPodWithContainer(f, ns, podname1, jig.Labels, []v1.Container{container1, container2})
 		names[podname1] = true
 		validateEndpointsPortsOrFail(cs, ns, serviceName, portsByPodName{podname1: {port1}})
```

Comment:
at this point we are only validating for 1 port, however the pod has 2 ports.
This is a good catch and a variation of current test,  so we have to add a new test with this scenario.

For this test the thing goes:
Create a service with multiple ports, targeting port1 and port2
Create pod1 exposing port1
Validate that endpoints are created with ONLY port1
--> we should check here that traffic also works and reach pod1
Create pod2 exposing port2
Validate that endpoints have now BOTH ports
--> we should check here that traffic reaches pod1 and pod2

It seems that CheckServiceReachability expects that ports are exposed in ALL pods, that is why it fails.
Maybe we can't use it and add our own polling logic here?